contracts/OffsetHelper.sol

```diff
@@ -57,66 +56,102 @@ import "./interfaces/IToucanContractRegistry.sol";
-     * contract. These can be modified after deployment by the contract owner
-     * using `setEligibleTokenAddress()` and `deleteEligibleTokenAddress()`.
-     *
-     * @param _eligibleTokenSymbols A list of token symbols.
-     * @param _eligibleTokenAddresses A list of token addresses corresponding
-     * to the provided token symbols.
-     */
-    constructor(
-        string[] memory _eligibleTokenSymbols,
-        address[] memory _eligibleTokenAddresses
-    ) {
-        uint256 i = 0;
-        uint256 eligibleTokenSymbolsLen = _eligibleTokenSymbols.length;
-        while (i < eligibleTokenSymbolsLen) {
-            eligibleTokenAddresses[
-                _eligibleTokenSymbols[i]
-            ] = _eligibleTokenAddresses[i];
-            i += 1;
-        }
-    }
+    // Chain ID of Celo mainnet
+    uint256 private constant CELO_MAINNET_CHAINID = 42220;
 
     /**
      * @notice Emitted upon successful redemption of TCO2 tokens from a Toucan
-     * pool token such as BCT or NCT.
+     * pool token e.g., NCT.
      *
-     * @param who The sender of the transaction
+     * @param sender The sender of the transaction
      * @param poolToken The address of the Toucan pool token used in the
-     * redemption, for example, NCT or BCT
+     * redemption,  e.g., NCT
      * @param tco2s An array of the TCO2 addresses that were redeemed
      * @param amounts An array of the amounts of each TCO2 that were redeemed
      */
     event Redeemed(
-        address who,
+        address sender,
         address poolToken,
         address[] tco2s,
         uint256[] amounts
     );
 
     modifier onlyRedeemable(address _token) {
-        require(isRedeemable(_token), "Token not redeemable");
+        require(isRedeemable(_token), "Token not redeemable.");
 
         _;
     }
 
     modifier onlySwappable(address _token) {
-        require(isSwappable(_token), "Token not swappable");
+        require(isSwappable(_token), "Path doesn't yet exist.");
 
         _;
     }
 
+    modifier isCelo() {
+        require(
+            block.chainid != CELO_MAINNET_CHAINID,
+            "The function is not available on Celo."
+        );
+
+        _;
+    }
+
+    /**
+     * @notice Contract constructor. Should specify arrays of ERC20 symbols and
+     * addresses that can used by the contract.
+     *
+     * @dev See `isEligible()` for a list of tokens that can be used in the
+     * contract. These can be modified after deployment by the contract owner
+     * using `setEligibleTokenAddress()` and `deleteEligibleTokenAddress()`.
+     *
+     * @param _poolAddresses A list of pool token addresses.
+     * @param _tokenSymbolsForPaths An array of symbols of the token the user want to retire carbon credits for
+     * @param _paths An array of arrays of addresses to describe the path needed to swap form the baseToken to the pool Token
+     * to the provided token symbols.
+     */
+    constructor(
```

Comment:
If this contract is meant to be upgradeable and run as a proxy, then you need to move all constructor logic inside initialize(). Otherwise, if the contract is meant to be immutable, I'd replace the OwnableUpgradeable contract with a Ownable contract.

Reply:
the next step for the OffsetHelper will be to remove all functions related to balance and deposit so, that it's not possible for the OffsetHelper to hold funds. And then we wouldn't need an Upgradable function necessarily, right?

Reply:
Yeah, I am not sure why upgradeability was added in the first place, I remember this used to be an immutable contract. Anyway, as it stands the `constructor` code will not run in the proxy. You are planning to use the proxy, right? Otherwise, I'd recommend removing upgradeability altogether.

Reply:
I see it's removed in the last commit. All good then!